### DISCUSSION TITLE
Welcome to Tasmotizer Discussions!

### DISCUSSION THREAD

**Discussion by jziolkowski** [tasmota/tasmotizer#68] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with Tasmotizer users. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.

**Comment by rin67630** [comment]
Can one use Tasmotizer together with code that was compiled with the Arduino IDE?
If yes, what are the preconditions to enable the WiFi parameters injection?
Thank you...

**Reply by jziolkowski** [reply]
I have completely zero idea what you are talking about

**Reply by rin67630** [reply]
I must admit, that I haven't an idea what Tasmotizer really is neither.
Apparently it is not an uploader for one's own compiled code, is it?

**Reply by jziolkowski** [reply]
It certainly is. But it uses esptool settings specific to work with tasmota. Your custom code might work, but there's no warranty.
As for the wifi settings etc: again, this mechanism uses tasmota commands, so you would have to basically mimic the same commands on serial interface.

**Comment by Bob1946** [comment]
Will there ever be a Tasmotizer for Linux as I can't find any instructions on YouTube or the web and I sometimes use Linux If anyone can help by pointing me to instructions that would be good I am new to programming so Tasmotizer was a blessing and I don't use Home assistant.
Thanks, anyone

**Reply by jziolkowski** [reply]
There is already, use option 2 or 3.

**Comment by lollo9710** [comment]
Good morning,

I am new to this comunity a greeting goes to all the developers of Tasmota.
I wanted to ask if I first came out a tool for

https://it.aliexpress.com/item/1005003295559182.html?spm=a2g0o.productlist.0.0.38bb6125rBO7At&algo_pvid=c3bb3ebb-aca8-4bb1-821c-2579b62dfd0c&algo_exp_id=c3bb3ebb-aca8-4bb1-821c-2579b62dfd0c-6&pdp_ext_f=%7B%22sku_id%22%3A%2212000025071710822%22%7D


Do you know if it will ever come out?

**Comment by jziolkowski** [comment]
Hi. This seems to be a generic ESP12 board; you should be able to flash it using an UART adapter, recommended one is based on a CH340 chip.

**Reply by lollo9710** [reply]
no è una scheda di queste caratteristiche:

Relè di WIFI del regolatore del modulo di WIFI di ESP-12F del modulo di WIFI di cc 5V 12V 24V ESP8266 ESP-12F 16 canali
con a bordo di due registri a scorrimento 74HC595 interruttore del relè di controllo.

**Reply by jziolkowski** [reply]
Please use english.

Either way, like I wrote, the module contains a completely exposed ESP-12F, so if there are no RX/TX/GPIO0 headers available, you can easily solder to ESP and flash away

**Comment by Thomas1305** [comment]
Hi, i ordered this board, 

https://it.aliexpress.com/item/1005003295559182.html?spm=a2g0o.productlist.0.0.38bb6125rBO7At&algo_pvid=c3bb3ebb-aca8-4bb1-821c-2579b62dfd0c&algo_exp_id=c3bb3ebb-aca8-4bb1-821c-2579b62dfd0c-6&pdp_ext_f=%7B%22sku_id%22%3A%2212000025071710822%22%7D

but i dont know how to control it with tasmota.
The board hast to shift register (74HC595) on it, which makes it possible to control 16 relays with it.
But i have no glue how to.

Can someone help me?
Best regards

**Reply by TallyMih** [reply]
you did it?
![64bb7620-bd46-4c68-bb90-130175d7eb1f](https://user-images.githubusercontent.com/24877431/210262132-9b041217-e9f2-4ac8-8911-355b45aa54fd.jpg)
me i have with 8 chanels

**Comment by jziolkowski** [comment]
It's not a tasmotizer issue, but tasmota.

Support for this board has been added few months ago, but it requires compiling your own software.

EDIT: i'll post a config template today or tomorrow on blakadder's template page

**Reply by Thomas1305** [reply]
Thank you so much.
Can you pls put a link to "blaladder´s template page". 
I still have tasmota 11.0.0 on the board and later one i wanna control it with iobroker.

**Reply by jziolkowski** [reply]
https://templates.blakadder.com/ but the device in question is not there yet

**Reply by Thomas1305** [reply]
ok, i´ll wait.
thx

**Reply by jziolkowski** [reply]
Didn't have time to make the template page yet, but here is what you need:

Like I said, you need to compile your own tasmota, with required flags enabled in user_config_override as shown here:

https://github.com/arendst/Tasmota/blob/development/tasmota/my_user_config.h#L914-L916

For this board, there are 2 shift registers. The invert flag can be ignored or set to false.

Then, this is the template for you:

`{"NAME":"16Relays","GPIO":[1,1,1,1,1,7712,1,1,7680,7648,7744,1,1,1],"FLAG":0,"BASE":18}`

**Reply by jziolkowski** [reply]
Also bear in mind, that there are few (most likely unimportant issues), f.e. tasmota autodiscovery reports at most only 8 first relays, same goes with relay's friendlynames

**Reply by jziolkowski** [reply]
Yes, you need to compile your own build, without that (as you can see) the template does nothing.

As for iobroker, I have no idea, never used it.

**Reply by Thomas1305** [reply]
ok, maybe if i first compile my build, than use the template i will see the switches on the tasmota surface

**Reply by jziolkowski** [reply]
yes you will, I see all 16 buttons on my compiled build

**Reply by Thomas1305** [reply]
ok, thats takes me lot of time now :-)
unbelievable what you forget over time

**Reply by Thomas1305** [reply]
or can you compile it for me?

sorry for the cheeky question

**Reply by Thomas1305** [reply]
i gues i just have to restet the wifi

**Reply by jziolkowski** [reply]
sorry, don't have time for compiling it now. You can use one of the methods available in tasmota docs.

**Reply by Thomas1305** [reply]
yes, it was just ja stupid question :-)
and i have only to change in line 914 - 916 in my user config.h ?

**Reply by Thomas1305** [reply]
right?
![image](https://user-images.githubusercontent.com/46224531/160873786-ac380d8d-360b-481c-9e37-a7bbcebd4825.png)

**Reply by jziolkowski** [reply]
no, in the user_config_override. 

Providing support for compilation is outside of the scope of this project anyway. Please refer to tasmota docs and/or tasmota discord.

**Reply by Thomas1305** [reply]
ok, i´ll try my best :-p

**Reply by Thomas1305** [reply]
damn
what am I doing wrong
![image](https://user-images.githubusercontent.com/46224531/161041270-da5596df-0638-4a93-8b39-f7a20c687e73.png)

**Reply by Thomas1305** [reply]
looks better
![image](https://user-images.githubusercontent.com/46224531/161044266-3a78c19c-952a-4bbc-a221-d7fae2e898d0.png)

**Reply by jziolkowski** [reply]
you didn't enable the main flag. You need to uncomment 

`//#define USE_SHIFT595`

**Reply by Thomas1305** [reply]
this way? 
![image](https://user-images.githubusercontent.com/46224531/161047123-6613a5d5-2b5d-4c3a-a2a5-62fd869d9834.png)

**Reply by Thomas1305** [reply]
i´m getting closer
![image](https://user-images.githubusercontent.com/46224531/161047886-d3456071-7d16-475a-a596-3ee99192d4d1.png)

**Reply by Thomas1305** [reply]
relay 1 and 9 switch at the same time
and so on
2 and 10 ..... 
8 and 16

**Reply by Thomas1305** [reply]
damn, forgot the template :-(

sorry. tamplate was still correct

**Reply by jziolkowski** [reply]
try this command in console:

`Shitf595DeviceCount 2`

**Reply by Thomas1305** [reply]
now it looks really good :-)

![image](https://user-images.githubusercontent.com/46224531/161048915-729d4b1d-6ed5-4473-8c43-0150910c7b79.png)

**Reply by Thomas1305** [reply]
i think he did not set 
 #define SHIFT595_DEVICE_COUNT  2                 // [Shift595DeviceCount] Set the number of connected 74x595 shift registers

**Reply by Thomas1305** [reply]
got it
there are two spaces before 2

**Reply by Thomas1305** [reply]
thank you so much

**Reply by jziolkowski** [reply]
no problem, glad it works. Which jalousien are you connecting to it?

**Reply by Thomas1305** [reply]
there are 8 external blinds in our house, which are manually controlled with a switch. in the future i would like to be able to control it automatically and  with my mobile phone 
this is the installed controller
![2b3c241e-2de1-46ac-bb97-74eed367f952](https://user-images.githubusercontent.com/46224531/161051493-a817037b-68ce-49b5-a561-c3233bf7cda7.jpg)

**Reply by Thomas1305** [reply]
and by the way, it´s working on the iobroker too
![image](https://user-images.githubusercontent.com/46224531/161051810-e31e9e6d-9428-4f50-8627-d26ab2625847.png)

**Reply by jziolkowski** [reply]
Nice setup. Remember that when connecting external blinds to these cheap relays that you see on this module with ESP, you need to add some external protection (varistors, for example) otherwise arcing will fry the relay contacts

**Comment by sparkgap** [comment]
**[Tasmotizer-1.2 run, but get no IP address](https://community.openhab.org/t/tasmotizer-1-2-run-but-get-no-ip-address/162539)**
New to Tasmoitizer. After running “Tasmotizer”, (Release 14.4.1), and using “Send config”, I get no IP address when clicking green “Get IP” button. Yes… DHCP is enabled in my LynkSys router. My previous method, using the Arduino IDE does provide an IP.  Help Please!

**Comment by ghost** [comment]
I try to flash a LSC smart power plug with tasmotizer., with TYWE2S module.
Where can I find the bib-file for this smart power plug ? (LSC 970762)
In the list of available wifi networks, I can see the TYWE2S module with SSID : 'tasmoto-............'
When I connect to this wifi network, and go with the browser to 192.168.4.1, Ihave the start screen of tasmota. Unfortunataly I always receive an error message that the module cannot to then network ( SSID and password are correct)
What do I wrong ? 
<img width="471" height="590" alt="ScreenShot_5_08_2025_23_52_28" src="https://github.com/user-attachments/assets/c527ae50-7993-4294-91c6-758a317f54c9" />
